crates/egui/src/text_selection/visuals.rs

```diff
@@ -96,8 +96,17 @@ pub fn paint_text_selection(
 pub fn paint_cursor_end(painter: &Painter, visuals: &Visuals, cursor_rect: Rect) {
     let stroke = visuals.text_cursor.stroke;
 
-    let top = cursor_rect.center_top();
-    let bottom = cursor_rect.center_bottom();
+    let (top, bottom) = if (stroke.width as usize) % 2 == 0 {
+        (
+            painter.round_pos_to_pixels(cursor_rect.center_top()),
+            painter.round_pos_to_pixels(cursor_rect.center_bottom()),
+        )
+    } else {
+        (
+            painter.round_pos_to_pixel_center(cursor_rect.center_top()),
+            painter.round_pos_to_pixel_center(cursor_rect.center_bottom()),
+        )
+    };
```

Comment:
I would think it's safe to assume that users will be using either 1px or 2px cursor thickness.

Reply:
Will this work when the scale is 150%?

Also it'd be nice to have a screenshot test that shows the cursor, maybe you can add one?